core/src/test/java/kafka/admin/UserScramCredentialsCommandTest.java

```diff
@@ -0,0 +1,184 @@
+            quotaPossibilityBOut + scramPossibilityBOut);
+
+        // now delete configs, in opposite order, for user1 and user2, and describe
+        deleteConfig(USER1, "consumer_byte_rate");
+        deleteConfig(USER2, "SCRAM-SHA-256");
+        describeUsers(quotaMessage(USER2) + describeUserMessage(USER1));
+
+        // now delete the rest of the configs, for user1 and user2, and describe
+        deleteConfig(USER1, "SCRAM-SHA-256");
+        deleteConfig(USER2, "consumer_byte_rate");
+        describeUsers("");
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"kraft", "zk"})
+    public void testAlterWithEmptyPassword(String quorum) {
+        String user1 = "user1";
+        ConfigCommandResult result = runConfigCommandViaBroker("--user", user1, "--alter", "--add-config", "SCRAM-SHA-256=[iterations=4096,password=]");
+        assertTrue(result.exitStatus.isPresent(), "Expected System.exit() to be called with an empty password");
+        assertEquals(1, result.exitStatus.getAsInt(), "Expected empty password to cause failure with exit status=1");
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"kraft", "zk"})
+    public void testDescribeUnknownUser(String quorum) {
+        String unknownUser = "unknownUser";
+        ConfigCommandResult result = runConfigCommandViaBroker("--user", unknownUser, "--describe");
+        assertFalse(result.exitStatus.isPresent(), "Expected System.exit() to not be called with an unknown user");
+        assertEquals("", result.stdout);
+    }
+
+    private void createAndAlterUser(String user) throws InterruptedException {
+        // create and describe a credential
+        ConfigCommandResult result = runConfigCommandViaBroker("--user", user, "--alter", "--add-config", "SCRAM-SHA-256=[iterations=4096,password=foo-secret]");
+        assertEquals(updateUserMessage(user), result.stdout);
+        TestUtils.waitUntilTrue(
+            () -> Objects.equals(runConfigCommandViaBroker("--user", user, "--describe").stdout, describeUserMessage(user)),
+            () -> "Failed to describe SCRAM credential change '" + user + "'");
+        // create a user quota and describe the user again
+        result = runConfigCommandViaBroker("--user", user, "--alter", "--add-config", "consumer_byte_rate=20000");
+        assertEquals(updateUserMessage(user), result.stdout);
+        TestUtils.waitUntilTrue(
+            () -> Objects.equals(runConfigCommandViaBroker("--user", user, "--describe").stdout, quotaMessage(user) + describeUserMessage(user)),
+            () -> "Failed to describe Quota change for '" + user + "'");
+    }
+
+    private void deleteConfig(String user, String config) {
+        ConfigCommandResult result = runConfigCommandViaBroker("--user", user, "--alter", "--delete-config", config);
+        assertEquals(updateUserMessage(user), result.stdout);
+    }
+
+    private void describeUsers(String... msgs) throws InterruptedException {
+        TestUtils.waitUntilTrue(
```

Comment:
We can throw NoRetryException instead of adding new method. For example:
```java
        TestUtils.waitForCondition(
            () -> {
                try {
                    String output = runConfigCommandViaBroker("--entity-type", "users", "--describe").stdout;
                    return Arrays.asList(msgs).contains(output);
                } catch (Exception e) {
                    throw new NoRetryException(e);
                }
            },
            () -> "Failed to describe config");
```

Reply:
Fixed.

Reply:
@chia7712 sorry I had suggested otherwise. To help me make better decisions in the future, can you share why it's preferrable not to have a the method?

Reply:
> To help me make better decisions in the future, can you share why it's preferrable not to have a the method?

The main reason is that I try to reduce the size of `TestUtils`. We are used to put something test-specified code to `TestUtils`, and that causes a fat/chaos class file. In fact, we just do a cleanup for `TestUtils` (#15808).

If re-throwing `NoRetryException` is too complicated, we can add a helper method to `UserScramCredentialsCommandTest`.

Reply:
I'd rather keep at as is.
Looks like @soarez agreed as PR approved by him :)

Reply:
Agreed. Thanks for the clarification 🙏